datatree/treenode.py

```diff
@@ -643,12 +668,12 @@ def _path_to_ancestor(self, ancestor: NamedNode) -> NodePath:
             raise NotFoundInTreeError(
                 "Cannot find relative path to ancestor because nodes do not lie within the same tree"
             )
-        if ancestor.path not in list(a.path for a in self.ancestors):
+        if ancestor.path not in list(a.path for a in (self, *self.parents)):
             raise NotFoundInTreeError(
                 "Cannot find relative path to ancestor because given node is not an ancestor of this node"
             )
 
-        lineage_paths = list(ancestor.path for ancestor in self.lineage)
-        generation_gap = list(lineage_paths).index(ancestor.path)
-        path_upwards = "../" * generation_gap if generation_gap > 0 else "/"
+        parents_paths = list(parent.path for parent in (self, *self.parents))
+        generation_gap = list(parents_paths).index(ancestor.path)
+        path_upwards = "../" * generation_gap if generation_gap > 0 else "."
```

Comment:
Same, this path was causing me issues, but the `else` branch might never be reached, for the same reason. However, using a `dot` fixed some tests during development. It makes sense: the path from a node itself is not the root (`/`) but `.` as declared in existing `test_relative_paths`: 

https://github.com/xarray-contrib/datatree/blob/66397e89529b2a3ecaa84d4945bd00da12a573a7/datatree/tests/test_datatree.py#L98